pkg/controllers/internalmembercluster/v1beta1/member_controller_test.go

```diff
@@ -639,3 +653,528 @@ func TestReportClusterPropertiesWithPropertyProvider(t *testing.T) {
+					Spec: corev1.PodSpec{
+						NodeName: nodeName3,
+						Containers: []corev1.Container{
+							{
+								Name: containerName1,
+								Resources: corev1.ResourceRequirements{
+									Requests: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse("1"),
+										corev1.ResourceMemory: resource.MustParse("1Gi"),
+									},
+								},
+							},
+						},
+					},
+					Status: corev1.PodStatus{
+						Phase: corev1.PodRunning,
+					},
+				},
+			},
+			wantIMCStatus: clusterv1beta1.InternalMemberClusterStatus{
+				Properties: map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
+					propertyprovider.NodeCountProperty: {
+						Value: "3",
+					},
+				},
+				ResourceUsage: clusterv1beta1.ResourceUsage{
+					Capacity: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("31"),
+						corev1.ResourceMemory: resource.MustParse("24Gi"),
+					},
+					Allocatable: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("25"),
+						corev1.ResourceMemory: resource.MustParse("19Gi"),
+					},
+					Available: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("16.5"),
+						corev1.ResourceMemory: resource.MustParse("9Gi"),
+					},
+				},
+			},
+		},
+		{
+			name:  "with pod not yet scheduled",
+			nodes: nodes,
+			pods: []*corev1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: podName1,
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name: containerName1,
+								Resources: corev1.ResourceRequirements{
+									Requests: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse("2"),
+										corev1.ResourceMemory: resource.MustParse("1Gi"),
+									},
+								},
+							},
+							{
+								Name: containerName2,
+								Resources: corev1.ResourceRequirements{
+									Requests: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse("1"),
+										corev1.ResourceMemory: resource.MustParse("2Gi"),
+									},
+								},
+							},
+						},
+					},
+					Status: corev1.PodStatus{
+						Phase: corev1.PodRunning,
+					},
+				},
+			},
+			wantIMCStatus: clusterv1beta1.InternalMemberClusterStatus{
+				Properties: map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
+					propertyprovider.NodeCountProperty: {
+						Value: "3",
+					},
+				},
+				ResourceUsage: clusterv1beta1.ResourceUsage{
+					Capacity: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("31"),
+						corev1.ResourceMemory: resource.MustParse("24Gi"),
+					},
+					Allocatable: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("25"),
+						corev1.ResourceMemory: resource.MustParse("19Gi"),
+					},
+					Available: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("25"),
+						corev1.ResourceMemory: resource.MustParse("19Gi"),
+					},
+				},
+			},
+		},
+		{
+			name:  "with pod succeeded",
```

Comment:
a bit more descriptive name for the test would be nice

Reply:
Fixed.